src/main/java/edu/ucsb/ucsbcslas/controllers/TutorAssignmentController.java

```diff
@@ -221,55 +235,109 @@
+        Optional<TutorAssignment> tutorAssignment = tutorAssignmentRepository.findById(id);
+        if (!tutorAssignment.isPresent()) {
+            return ResponseEntity.notFound().build();
         }
-      } else {
+        logger.info(incomingTutorAssignment);
+        JSONObject ta = new JSONObject(incomingTutorAssignment);
+        TutorAssignment newAssignment = new TutorAssignment();
+        logger.info("ta: ", ta.toString());
+        JSONObject cInfo = new JSONObject(ta.get("course").toString());
+        logger.info("cInfo: ", cInfo.toString());
+        Course c = new Course(cInfo.getLong("id"), cInfo.getString("name"), cInfo.getString("quarter"),
+                cInfo.getString("instructorFirstName"), cInfo.getString("instructorLastName"),
+                cInfo.getString("instructorEmail"));
+        newAssignment.setCourse(c);
 
-        JSONObject tInfo = new JSONObject(ta.get("tutor").toString());
-        Tutor t = new Tutor(tInfo.getLong("id"), tInfo.getString("firstName"), tInfo.getString("lastName"), 
-        tInfo.getString("email"));
-        newAssignment.setTutor(t);
+        logger.info("tutorAssignment: ", tutorAssignment.get().getTutor().getEmail());
+        logger.info("ta: ", ta.getString("tutorEmail"));
+        if (!(ta.getString("tutorEmail").equals(tutorAssignment.get().getTutor().getEmail()))) {
+            Optional<Tutor> tutor = tutorRepository.findByEmail(ta.getString("tutorEmail"));
+            logger.info("tutor: ", tutor);
+            if (tutor.isPresent()) {
+                newAssignment.setTutor(tutor.get());
+            } else {
+                return getIncorrectInputResponse();
+            }
+        } else {
 
+            JSONObject tInfo = new JSONObject(ta.get("tutor").toString());
+            Tutor t = new Tutor(tInfo.getLong("id"), tInfo.getString("firstName"), tInfo.getString("lastName"),
+                    tInfo.getString("email"));
+            newAssignment.setTutor(t);
 
-      }
-      newAssignment.setAssignmentType(ta.getString("assignmentType"));
+        }
+        newAssignment.setAssignmentType(ta.getString("assignmentType"));
 
+        newAssignment.setId(id);
+        tutorAssignmentRepository.save(newAssignment);
+        String body = mapper.writeValueAsString(newAssignment);
+        return ResponseEntity.ok().body(body);
+    }
 
-      newAssignment.setId(id);
-      tutorAssignmentRepository.save(newAssignment);
-      String body = mapper.writeValueAsString(newAssignment);
-      return ResponseEntity.ok().body(body);
+    // Controller to upload CSV for tutorAssignment Service
+
+    @PostMapping(value = "/api/member/tutorAssignments/upload", produces = "application/json")
+    public ResponseEntity<String> uploadCSV(@RequestParam("csv") MultipartFile csv,
+            @RequestHeader("Authorization") String authorization) throws IOException {
+        logger.info("Starting upload CSV");
+        AppUser user = authControllerAdvice.getUser(authorization);
```

Comment:
We don't do anything with `user` there should be a check to see if `user` is of the correct role.

Reply:
A regular user can't see the upload tutor assignment option, which is why I decided to don't have to handle that case

Reply:
Even if the link to this page might be hidden from a regular user, they may still be able to access it directly via the url if we don't enforce authorization here

Reply:
I updated that. But there is a catch exception error test that I am not passing due to I changed some of the code (it was passing before I introduced an if statement) Is it possible to fix that on another  PR?

Reply:
Tests have been introduced to cover this case, marking conversation as resolved.